docker/scripts/setup_cnf.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash 
+
+mongodb1=`getent hosts ${MONGO1} | awk '{ print $1 }'`
+mongodb2=`getent hosts ${MONGO2} | awk '{ print $1 }'`
+mongodb3=`getent hosts ${MONGO3} | awk '{ print $1 }'`
+arbiter=`getent hosts ${ARBITER} | awk '{ print $1 }'`
+
+port=${PORT:-27017}
+
+echo "Waiting for startup.."
+until mongo --host ${mongodb1}:${port} --eval 'quit(db.runCommand({ ping: 1 }).ok ? 0 : 2)' &>/dev/null; do
+  printf '.'
+  sleep 1
+done
+
+echo "Started.."
+
+echo setup.sh time now: `date +"%T" `
+mongo --host ${mongodb1}:${port} <<EOF
```

Comment:
Since this is the only part different in both the scripts, I would prefer to reuse sections of the script.

We can abstract the replica set configuration part to a function. There can be one function for config server replica set and one for normal server replica set. They can be selected based on parameter passed to script.

Reply:
I've migrated both config scripts to the single one.